internal/gatewayapi/helpers.go

```diff
@@ -495,3 +496,37 @@ func irConfigName(policy client.Object) string {
 		strings.ToLower(policy.GetObjectKind().GroupVersionKind().Kind),
 		utils.NamespacedName(policy).String())
 }
+
+func getPolicyTargetRefs[T client.Object](policy egv1a1.PolicyTargetReferences, potentialTargets []T) []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName {
+	dedup := map[gwapiv1a2.LocalPolicyTargetReferenceWithSectionName]byte{}
+	for _, ref := range policy.GetTargetRefs() {
+		dedup[ref] = byte(0)
+	}
+
+	for _, currSelector := range policy.TargetSelectors {
```

Comment:
need to account for ReferenceGrant here

Reply:
Why?

Policies are not allowed to access targets across namespaces. These are all `LocalPolicyTargetReference` attachments.

Adding support for cross-namespace access for policies is a rather large change.

Reply:
ah thanks for reminding me, we only support same namespace today